grayskull/strategy/pypi.py

```diff
@@ -511,3 +545,49 @@ def remove_selectors_pkgs_if_needed(
             pkg = re_selector.sub("", pkg)
         result.append(pkg)
     return result
+
+
+def extract_optional_requirements(metadata: dict, config) -> Dict[str, List[str]]:
```

Comment:
can you create unit tests for this 3 functions here please?
After that I think that is good to go

Reply:
Sure

Reply:
Done.